crates/fsrt/src/main.rs

```diff
@@ -184,35 +109,25 @@ fn collect_sourcefiles<P: AsRef<Path>>(root: P) -> impl Iterator<Item = PathBuf>
         })
 }
 
+#[allow(dead_code)]
+pub(crate) fn scan_directory_test(
+    forge_test_proj: MockForgeProject<'_>,
+) -> Option<Vec<Vulnerability>> {
+    scan_directory(PathBuf::new(), &Args::parse(), forge_test_proj)
+}
+
 #[tracing::instrument(level = "debug")]
-fn scan_directory(dir: PathBuf, function: Option<&str>, opts: &Args) -> Result<()> {
-    let mut manifest_file = dir.clone();
-    manifest_file.push("manifest.yaml");
-    if !manifest_file.exists() {
-        manifest_file.set_extension("yml");
-    }
-    debug!(?manifest_file);
+pub(crate) fn scan_directory<'a>(
+    dir: PathBuf,
+    opts: &Args,
+    project: impl ForgeProjectTrait<'a> + std::fmt::Debug,
+) -> Option<Vec<Vulnerability>> {
```

Comment:
```suggestion
) -> Result<Vec<Vulnerability>> {
```

That way, we can test if we get correct errors, and you don't have to keep calling `.ok()` on all of the returned errors to convert them to options.

Reply:
Fixed. Thanks :)